OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationBundleProcessor.java

```diff
@@ -112,51 +112,63 @@ static void processFromFCMIntentService(Context context, BundleCompat bundle, OS
 
     /**
      * Recommended method to process notification before displaying
-     * Only use the {@link NotificationBundleProcessor#processJobForDisplay(OSNotificationGenerationJob, boolean, boolean)}
+     * Only use the {@link NotificationBundleProcessor#processJobForDisplay(OSNotificationGenerationJob, boolean)}
      *  in the event where you want to mark a notification as opened or displayed different than the defaults
      */
     @WorkerThread
-    static int processJobForDisplay(OSNotificationGenerationJob notificationJob) {
-        return processJobForDisplay(notificationJob, false, true);
+    static int processJobForDisplay(OSNotificationGenerationJob notificationJob, boolean callForegroundLogic) {
+        OSNotificationController notificationController = new OSNotificationController(notificationJob, notificationJob.isRestoring(), true);
+        return processJobForDisplay(notificationController, false, callForegroundLogic);
```

Comment:
Follow up from this comment
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1156#pullrequestreview-495057331

This also should be renamed, as this `callForegroundLogic` ends up becoming `fromBackgroundLogic` in the `processJobForDisplay(OSNotificationController notificationController, boolean opened, boolean fromBackgroundLogic)` below

Reply:
good catch!